build-logic/code-quality/src/main/kotlin/testng.errorprone.gradle.kts

```diff
@@ -0,0 +1,15 @@
+import net.ltgt.gradle.errorprone.errorprone
+
+plugins {
+    id("net.ltgt.errorprone")
+    id("build-logic.build-params")
+}
+
+dependencies {
+    errorprone("com.google.errorprone:error_prone_core:2.20.0")
+}
+
+tasks.withType<JavaCompile>().configureEach {
+    options.errorprone.isEnabled = !buildParameters.skipErrorProne
```

Comment:
A slightly better option would be avoid verifying `buildParameters.skipErrorProne` and go with conditional apply of `testng.errorprone` plugin

Reply:
Here's an example: https://github.com/pgjdbc/pgjdbc/blob/93b0fcb2711d9c1e3a2a03134369738a02a58b40/build-logic/verification/src/main/kotlin/build-logic.style.gradle.kts#L44-L46

However, both ways work.
The slight difference is as follows: imagine there's an issue with errorprone plugin. For instance, imagine errorprone fails with NPE. If you completely skip `net.ltgt.errorprone` then it would likely allow to avoid failures. Of course, `errorprone.isEnabled` might work, however, it might fail to work :)

Reply:
Thanks for your feedback.
I will update